### ISSUE TITLE
Applicative-Alternative agreement law

### ISSUE THREAD

**Issue by vrom911** [kowainik/validation-selective#35]
```haskell
empty <*> f = empty
```

See here: https://wiki.haskell.org/Typeclassopedia#Laws_6

**Comment by chshersh** [comment]
Apparently, `Alternative` instance for `Validation` doesn't satisfy this law, because:

```haskell
empty <|> Failure "foo" = Failure "foo"
```

But I think that this is fine. The connection between `Applicative` and `Alternative` is not officially documented, and all laws I saw are just people thinking what would be the reasonable set of laws.

There's a weaker typeclass called `Alt` in the `semigroupoids` package, and some people believe that standard hierarchy should be refactored to use it.

```haskell
-- like Alternative but without empty
class Functor f => Alt f where
    (<!>) :: f a -> f a -> f a
```

In short, if typeclass hierarchy was more modular, we could implement lawful `Alt` instance but not `Alternative`. Still, for historical reasons, we're using `Alternative`, and I genuinely think this is fine.
